examples/hosts/app-integration/corruption-recovery/src/app.tsx

```diff
@@ -0,0 +1,150 @@
+
+import { AppView } from "./appView";
+
+// Define the schema of our Container.
+const containerSchema = {
+    initialObjects: {
+        dataMap: SharedMap,
+        recoveryMap: SharedMap,
+    },
+};
+
+async function initializeData(container: IFluidContainer): Promise<void> {
+    const sharedMap = container.initialObjects.dataMap as SharedMap;
+    await Promise.all([DataController.initializeModel(sharedMap)]);
+}
+
+async function createRecoveryAgent(
+    container: IFluidContainer,
+    containerId: string,
+): Promise<void> {
+    const rollContainerId = await RecoveryAgent.createRecoveryAgent(
+        containerId,
+    );
+    const recMap = container.initialObjects.recoveryMap as SharedMap;
+    recMap.set("recoverContainerId", rollContainerId);
+}
+
+function cleanRecoveryAgent(container: IFluidContainer): void {
+    const recMap = container.initialObjects.recoveryMap as SharedMap;
+    recMap.set("recoverContainerId", "");
+}
+
+async function getRecoveryAgent(
+    container: IFluidContainer,
+): Promise<RecoveryAgent | undefined> {
+    const recMap = container.initialObjects.recoveryMap as SharedMap;
+    let recoveryContainerId = recMap.get<string>("recoverContainerId");
+
+    if (!(recoveryContainerId ?? "")) {
+        await new Promise<void>((resolve) =>
+            recMap.once("valueChanged", () => resolve()),
+        );
+        recoveryContainerId = recMap.get("recoverContainerId");
+    }
+
+    return RecoveryAgent.getRecoveryAgent(
+        recoveryContainerId ?? "",
+        containerSchema,
+    );
+}
+
+async function start(): Promise<void> {
+    const clientProps = {
+        connection: connectionConfig,
+    };
+    const client = new AzureClient(clientProps);
+    let container: IFluidContainer;
+    let containerId: string;
+
+    const div = document.getElementById("content") as HTMLDivElement;
+    ReactDOM.render(
+        <div className="d-flex justify-content-center m-5">
+            <div className="spinner-border" role="status" />
+        </div>,
+        div,
+    );
+
+    // Get or create the document depending if we are running through the create new flow
+    const createNew = location.hash.length === 0;
+    if (createNew) {
+        // The client will create a new detached container using the schema
+        // A detached container will enable the app to modify the container before attaching it to the client
+        ({ container } = await client.createContainer(containerSchema));
+
+        // Initialize our models so they are ready for use with our controllers
+        await initializeData(container);
+
+        // If the app is in a `createNew` state, and the container is detached, we attach the container.
+        // This uploads the container to the service and connects to the collaboration session.
+        containerId = await container.attach();
+
+        // Setup recovery agent
+        await createRecoveryAgent(container, containerId);
```

Comment:
What happens container moves? I.e. equivalent to "file rename" happens? Maybe it's not supported today, but one can't assume it will never be possible.